fork_choice/confirmation-rule.md

```diff
@@ -0,0 +1,673 @@
+            # This block is not in the finalized chain.
+            return False
+        else:
+            # Check is_one_confirmed for this block and is_lmd_confirmed for the preceding chain.
+            return (
+                is_one_confirmed(store, confirmation_byzantine_threshold, block_root) and
+                is_lmd_confirmed(store, confirmation_byzantine_threshold, block.parent_root)
+            )
+```
+
+#### `get_total_active_balance_for_block_root`
+
+```python
+def get_total_active_balance_for_block_root(store: Store, block_root: Root) -> Gwei:
+    assert block_root in store.block_states
+
+    # Alternatively, fetch the state from the block's checkpoint
+    state = store.block_states[block_root]
+
+    return get_total_active_balance(state)
+```
+
+#### `get_remaining_weight_in_current_epoch`
+
+```python
+def get_remaining_weight_in_current_epoch(store: Store, block_root: Root) -> Gwei:
+    """ Returns the total weight of votes for this epoch from future committees after the current slot
+    """
+    assert block_root in store.block_states
+
+    # Alternatively, fetches the state from the block's checkpoint
+    state = store.block_states[block_root]
+
+    current_slot = get_current_slot(store)
+    first_slot_next_epoch = compute_start_slot_at_epoch(Epoch(compute_epoch_at_slot(current_slot) + 1))
+    return get_committee_weight_between_slots(state, Slot(current_slot + 1), Slot(first_slot_next_epoch - 1))
+```
+
+#### `get_leaf_block_roots`
+
+```python
+def get_leaf_block_roots(store: Store, block_root: Root) -> Set[Root]:
+    children = [
+        root for root in store.blocks.keys()
+        if store.blocks[root].parent_root == block_root
+    ]
+
+    if any(children):
+        # Get leaves of all children and add to the set.
+        leaves = set().union(*[get_leaf_block_roots(store, child) for child in children])
+        return leaves
+    else:
+        # This block is a leaf.
+        return set([block_root])
+
+```
+
+#### `get_current_epoch_participating_indices`
+
+```python
+def get_epoch_participating_indices(
+    state: BeaconState, 
+    active_validator_indices: Sequence[ValidatorIndex],
+    current_epoch: bool
+) -> Set[ValidatorIndex]: 
+    if current_epoch:
+        epoch_participation = state.current_epoch_participation
+    else:
+        epoch_participation = state.previous_epoch_participation
+
+    return set([
+        i for i in active_validator_indices 
+        if has_flag(epoch_participation[i], TIMELY_TARGET_FLAG_INDEX)
+    ])
+```
+
+#### `get_ffg_support`
+
+```python
+def get_ffg_support(store: Store, block_root: Root) -> Gwei:
+    """
+    Returns the total weight supporting the checkpoint in the block's chain at block's epoch
+    """
+
+    block = store.blocks[block_root]
+    block_epoch = compute_epoch_at_slot(block.slot)
+    current_epoch = get_current_store_epoch(store)
+
+    assert current_epoch <= block_epoch + 1
+    assert block_epoch <= current_epoch
```

Comment:
```suggestion
    assert block_epoch <= current_epoch <= block_epoch + 1
```

Or maybe `assert current_epoch in [block_epoch, block_epoch + 1]`. I think that is actually clearer to the reader

Reply:
Done. (I will push it soon.)